src/tools/registry/gen_inc.c

```diff
@@ -1377,40 +1447,18 @@ int parse_var_array(FILE *fd, ezxml_t registry, ezxml_t superStruct, ezxml_t var
 			fortprintf(fd, "         call mpas_pool_get_dimension(newSubPool, 'index_%s', const_index)\n", varname_in_code);
 			fortprintf(fd, "      end if\n");
 			fortprintf(fd, "      if (const_index > 0) then\n", spacing);
-			if ( vardesc != NULL ) {
-				string = strdup(vardesc);
-				tofree = string;
-
-				token = strsep(&string, "'");
-				sprintf(temp_str, "%s", token);
-
-				while ( ( token = strsep(&string, "'") ) != NULL ) {
-					sprintf(temp_str, "%s''%s", temp_str, token);
-				}
-
-				free(tofree);
-
-				fortprintf(fd, "         call mpas_add_att(%s %% attLists(const_index) %% attList, 'long_name', '%s')\n", pointer_name_arr, temp_str);
-			}
-
-			if ( varunits != NULL ) {
-				string = strdup(varunits);
-				tofree = string;
-
-				token = strsep(&string, "'");
-				sprintf(temp_str, "%s", token);
 
-				while ( ( token = strsep(&string, "'") ) != NULL ) {
-					sprintf(temp_str, "%s''%s", temp_str, token);
+			for (attr = var_xml->attr; attr && *attr; attr+=2) {
+				// If the attr is "missing_value", ignore it and later on take
+				// the value from the var array.
+				if (strcmp(attr[0], "missing_value") == 0) {
+					printf("WARNING: Ignoring missing_value definition in variable defined under %s\n", vararrname);
```

Comment:
Just a suggestion, but we could add the name of the variable in question to this warning message, e.g.,
```
printf("WARNING: Ignoring missing_value attribute for var %s defined in var_array %s\n", varname, vararrname);
```
(I also modified the terminology a bit.)

Reply:
Agreed, change made